.github/workflows/release.yaml

```diff
@@ -41,22 +44,57 @@ jobs:
       - name: Push main
         if: github.ref == 'refs/heads/main'
         run: |
+          REV=$(git describe --long --tags --match='v*' --dirty 2>/dev/null || git rev-list -n1 HEAD)
+          GIT_COMMIT=$(git rev-parse HEAD)
+          BUILD_DATE=$(date -u -Iseconds)
+          PKG=github.com/cloudstack/cloudstack-csi-driver
+          LDFLAGS="-s -w -X ${PKG}/pkg/driver.driverVersion=${REV} -X ${PKG}/pkg/driver.gitCommit=${GIT_COMMIT} -X ${PKG}/pkg/driver.buildDate=${BUILD_DATE}"
+          
           for img in $IMAGES; do
-              docker tag ${img} ${REGISTRY_NAME}/${img}:main
-              docker push ${REGISTRY_NAME}/${img}:main
+              docker buildx build \
+                --platform linux/amd64,linux/arm64 \
+                --file ./cmd/${img}/Dockerfile \
+                --build-arg LDFLAGS="${LDFLAGS}" \
+                --tag ${REGISTRY_NAME}/${img}:main \
+                --label org.opencontainers.image.revision=${GIT_COMMIT} \
+                --push \
+                .
           done
 
       - name: Push tagged release
         if: startsWith(github.ref, 'refs/tags/v')
         run: |
           # Strip prefix from version
           VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,' | sed -e 's/^v//')
+          REV=$(git describe --long --tags --match='v*' --dirty 2>/dev/null || git rev-list -n1 HEAD)
+          GIT_COMMIT=$(git rev-parse HEAD)
+          BUILD_DATE=$(date -u -Iseconds)
+          PKG=github.com/cloudstack/cloudstack-csi-driver
+          LDFLAGS="-s -w -X ${PKG}/pkg/driver.driverVersion=${REV} -X ${PKG}/pkg/driver.gitCommit=${GIT_COMMIT} -X ${PKG}/pkg/driver.buildDate=${BUILD_DATE}"
 
           for img in $IMAGES; do
-              docker tag ${img} ${REGISTRY_NAME}/${img}:${VERSION}
-              docker push ${REGISTRY_NAME}/${img}:${VERSION}
+              docker buildx build \
+                --platform linux/amd64,linux/arm64 \
+                --file ./cmd/${img}/Dockerfile \
+                --build-arg LDFLAGS="${LDFLAGS}" \
+                --tag ${REGISTRY_NAME}/${img}:${VERSION} \
+                --label org.opencontainers.image.revision=${GIT_COMMIT} \
+                --push \
+                .
           done
 
+      - name: Build syncer binary for upload
+        if: startsWith(github.ref, 'refs/tags/v')
+        run: |
+          REV=$(git describe --long --tags --match='v*' --dirty 2>/dev/null || git rev-list -n1 HEAD)
```

Comment:
Can we move all these params to a single step? this will make the github actions cleaner.